src/main/java/com/exasol/exaoperation/plugin/Plugin.java

```diff
@@ -87,14 +90,62 @@ public String getFileName() {
      */
     // [impl->dsn~installing-plug-ins~1]
     public ExecResult install() {
-        return runScript("install");
+        return callFunction("install");
     }
 
-    public ExecResult runScript(final String method) {
+    /**
+     * Call one of the plugin's functions.
+     * <p>
+     *     Plugins should handle function names case-insensitive.
+     * </p>
+     *
+     * @param method name of the function to call.
+     * @return result of the function call
+     */
+    public ExecResult callFunction(final String method) {
+        return callFunction(method, "");
+    }
+
+    /**
+     * Call one of the plugin's functions with the given argument.
+     * <p>
+     *     Plugins should handle function names case-insensitive; content and encoding of argument
+     *     depend on plugin and called function.
+     * </p>
+     *
+     * @param method   name of the function to call.
+     * @param argument argument to the function call.
+     * @return result of the function call
+     */
+    public ExecResult callFunction(final String method, final String argument) {
+        if (null == argument) {
+            throw new ExaOperationEmulatorException("Argument of Plugin::callFunction must never be null!");
+        }
+        return callFunctionInternal(method, argument);
+    }
+
+    /**
+     * Internal implementation of callFunction, distinguishing call with and without argument.
+     *
+     * @param method   Name of function to call
+     * @param argument optional single argument for the function call
+     * @return result of function call
+     */
+    private ExecResult callFunctionInternal(final String method, final String... argument) {
```

Comment:
Is it possible to move the contents of this function to above `callFunction`?
For me it seems they do not have to be separated.

Reply:
Yes they do, because otherwise listFunctions() does not have an entry point any more:
The public interface does not support calling functions without a parameter (see also the plugin_functions reference implementation in the installable plugin)

Reply:
`callFunction(String)` is just a convenience function I left in because many functions actually will not semantically require an argument.

Reply:
Thanks! Approved.